src/common/com/intellij/plugins/haxe/lang/psi/HaxeResolver.java

```diff
@@ -46,8 +46,12 @@
   public static final HaxeResolver INSTANCE = new HaxeResolver();
   public static final String IMPORT_EXTENSION = ".hx";
 
+  public static boolean isExtension = false;
```

Comment:
As a temporal solution we can use a ThreadLocal. So each thread has its own value. That should be safe and work too.